packages/gatsby/cache-dir/develop-static-entry.js

```diff
@@ -79,6 +91,109 @@ export default (pagePath, callback) => {
     postBodyComponents = components
   }
 
+  const getPageDataPath = path => {
+    const fixedPagePath = path === `/` ? `index` : path
+    return join(`page-data`, fixedPagePath, `page-data.json`)
+  }
+
+  const getPageData = pagePath => {
+    const pageDataPath = getPageDataPath(pagePath)
+    const absolutePageDataPath = join(process.cwd(), `public`, pageDataPath)
+    const pageDataRaw = fs.readFileSync(absolutePageDataPath)
+
+    try {
+      return JSON.parse(pageDataRaw.toString())
+    } catch (err) {
+      return null
+    }
+  }
+
+  const pageData = getPageData(pagePath)
+
+  const {
+    componentChunkName,
+    //  staticQueryHashes = []
+  } = pageData
+
+  const createElement = React.createElement
+
+  class RouteHandler extends React.Component {
+    render() {
+      const props = {
+        ...this.props,
+        ...pageData.result,
+        params: {
+          ...grabMatchParams(this.props.location.pathname),
+          ...(pageData.result?.pageContext?.__params || {}),
+        },
+        // pathContext was deprecated in v2. Renamed to pageContext
+        pathContext: pageData.result ? pageData.result.pageContext : undefined,
```

Comment:
```suggestion
        pathContext: pageData.result?.pageContext,
```

(With the subtle note that if it's relevant that `pageData.result` returns `null` if it is `null`, then this is an incorrect suggestion. Otherwise I think it's better.)

Reply:
I'm inclined to keep this the same as this code has been around for all of v2 & there's an off-chance that people explicitly check for `undefined`. We'd also need to make sure this code gets changed across static-entry.js & browser-entry.js